app/scripts/models/account.js

```diff
@@ -16,31 +16,41 @@ define([
   'lib/constants',
   'models/profile-image'
 ], function (Backbone, _, p, AuthErrors, ProfileClient, Constants, ProfileImage) {
-
   // Account attributes that can be persisted
   var PERSISTENT = {
-    uid: undefined,
+    accessToken: undefined,
     email: undefined,
+    grantedPermissions: undefined,
+    lastLogin: undefined,
+    needsOptedInToMarketingEmail: undefined,
+    profileImageId: undefined,
+    profileImageUrl: undefined,
     sessionToken: undefined,
     sessionTokenContext: undefined,
-    accessToken: undefined,
-    verified: undefined,
-    profileImageUrl: undefined,
-    profileImageId: undefined,
-    lastLogin: undefined,
-    grantedPermissions: undefined
+    uid: undefined,
+    verified: undefined
   };
 
   var DEFAULTS = _.extend({
-    password: undefined,
-    unwrapBKey: undefined,
+    customizeSync: undefined,
     keyFetchToken: undefined,
-    customizeSync: undefined
+    password: undefined,
+    unwrapBKey: undefined
   }, PERSISTENT);
 
   var ALLOWED_KEYS = Object.keys(DEFAULTS);
   var ALLOWED_PERSISTENT_KEYS = Object.keys(PERSISTENT);
 
+  function withMarketingEmailClient(method) {
+    return function () {
+      var self = this;
+      return self.getOAuthTokenFromSessionToken(self.get('sessionToken'), 'basket')
+        .then(function (accessToken) {
+          return self._marketingEmailClient[method](accessToken);
+        });
```

Comment:
For whatever reason, one thing that only occurred to me today is that I should probably call the client once to get all of the initial marketing email data, store that off into a model (either the account, or a separate email-preferences model that is related to the account), and use that one model to populate multiple fields on the `/settings/communications_preferences` page. Otherwise, two calls will be made to the backend to display that page, once for the user's opt-in preference, once for the user's Basket management URL.